src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,108 @@
-import { useState } from 'react';
+import React, { useEffect, useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd: (movie: Movie) => void;
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [enteredTitle, setEnteredTitle] = useState('');
+  const [enteredDescription, setEnteredDescription] = useState('');
+  const [enteredImageUrl, setEnteredImageUrl] = useState('');
+  const [enteredImdbUrl, setEnteredImdbUrl] = useState('');
+  const [enteredImdbId, setEnteredImdbId] = useState('');
+  const [formIsValid, setFormIsValid] = useState(false);
+  const [count, setCount] = useState(0);
+
+  const handleSumbit = () => {
+    const newEnteredMovie: Movie = {
+      title: enteredTitle,
+      description: enteredDescription,
+      imgUrl: enteredImageUrl,
+      imdbUrl: enteredImdbUrl,
+      imdbId: enteredImdbId,
+    };
+
+    onAdd(newEnteredMovie);
+
+    setEnteredTitle('');
+    setEnteredDescription('');
+    setEnteredImageUrl('');
+    setEnteredImdbUrl('');
+    setEnteredImdbId('');
+
+    setCount((state) => state + 1);
+  };
+
+  const isValidUrl = (urlString: string) => {
+    // eslint-disable-next-line max-len
+    const pattern = /^((([A-Za-z]{3,9}:(?:\/\/)?)(?:[-;:&=+$,\w]+@)?[A-Za-z0-9.-]+|(?:www\.|[-;:&=+$,\w]+@)[A-Za-z0-9.-]+)((?:\/[+~%/.\w-_]*)?\??(?:[-+=&;%@,.\w_]*)#?(?:[,.!/\\\w]*))?)$/;
+
+    return pattern.test(urlString);
+  };
+
+  useEffect(() => {
+    setFormIsValid(
+      isValidUrl(enteredImageUrl)
+      && isValidUrl(enteredImdbUrl)
+      && enteredTitle.length > 0
+      && enteredImdbId.length > 0
+      ,
+    );
+  }, [enteredTitle, enteredImageUrl, enteredImdbUrl, enteredImdbId]);
 
   return (
-    <form className="NewMovie" key={count}>
+    <form
+      className="NewMovie"
+      key={count}
+      onSubmit={(event) => {
+        event.preventDefault();
+
+        handleSumbit();
+      }}
```

Comment:
you could move preventDefault into handleSubmit and make it shorter here
```suggestion
      onSubmit={(event) => handleSumbit(event)};
      // or a short equivalent
      onSubmit={handleSumbit};
```